eclair-core/src/main/scala/fr/acinq/eclair/PluginParams.scala

```diff
@@ -55,4 +57,20 @@ trait CustomCommitmentsPlugin extends PluginParams {
    * returned by this method.
    */
   def getHtlcsRelayedOut(htlcsIn: Seq[IncomingHtlc], nodeParams: NodeParams, log: LoggingAdapter): Map[Origin, Set[(ByteVector32, Long)]]
+}
+
+// @formatter:off
+trait InterceptOpenChannelCommand
+case class InterceptOpenChannelReceived(replyTo: ActorRef[InterceptOpenChannelResponse], openChannelNonInitiator: OpenChannelNonInitiator, defaultParams: DefaultParams) extends InterceptOpenChannelCommand {
+  val remoteFundingAmount: Satoshi = openChannelNonInitiator.open.fold(_.fundingSatoshis, _.fundingAmount)
+  val temporaryChannelId: ByteVector32 = openChannelNonInitiator.open.fold(_.temporaryChannelId, _.temporaryChannelId)
+}
+
+sealed trait InterceptOpenChannelResponse
+case class AcceptOpenChannel(temporaryChannelId: ByteVector32, defaultParams: DefaultParams) extends InterceptOpenChannelResponse
+case class RejectOpenChannel(temporaryChannelId: ByteVector32, error: Error) extends InterceptOpenChannelResponse
+// @formatter:on
+
+trait InterceptOpenChannelPlugin extends PluginParams {
+  def openChannelInterceptor: ActorRef[InterceptOpenChannelCommand]
```

Comment:
I just noticed I used a `def` here, but it should be a `val`. Right?

Reply:
No, you should never use `val` in a `trait`, it should always be a `def` since the `trait` doesn't know how the concrete classes will implement this. The concrete class that implements this `trait` will most likely use a `val` though.